Sources/XMTPiOS/Client.swift

```diff
@@ -172,13 +215,40 @@ public final class Client: Sendable {
 			rustClient: client
 		)
 
-		return try Client(address: address, privateKeyBundleV1: v1Bundle, apiClient: apiClient)
+		let v3Client: FfiXmtpClient?
+
+		if options.enableAlphaMLS == true && options.api.env == .local {
+			let dbURL = URL.documentsDirectory.appendingPathComponent("xmtp-\(address).db3")
+			v3Client = try await LibXMTP.createClient(
+				logger: XMTPLogger(),
+				host: GRPCApiClient.envToUrl(env: apiClient.environment),
+				isSecure: apiClient.environment != .local,
+				db: dbURL.path,
+				encryptionKey: nil,
+				accountAddress: address,
+				legacyIdentitySource: .static,
+				legacySignedPrivateKeyProto: try v1Bundle.toV2().identityKey.serializedData()
+			)
+
+			try await v3Client?.registerIdentity(recoverableWalletSignature: nil)
```

Comment:
This is repeated it would probably be better to just do this in one place so that its harder to introduce bugs. Do you mind pulling this into it's own method?

Reply:
Done